ZOOKEEPER-3101: add reminder to ZOO_ERRORS

sl4mmy opened the pull request:
Add comment at bottom of ZOO_ERRORS reminding developers to also
update zerror(int) when adding or changing values.  Suggested by
@phunt in https://github.com/apache/zookeeper/pull/575#issuecomment-406356144.

Signed-off-by: Kent R. Spillner <kspillner@acm.org>

sl4mmy commented:
The Jenkins failure looks like an internal error in Jenkins itself: https://builds.apache.org/job/PreCommit-ZOOKEEPER-github-pr-build/1987/consoleText

sl4mmy commented:
This commit just adds a comment so I'm not sure how it would cause a build failure other than a style alert, but that doesn't appear to be the case

sl4mmy commented:
@nkalmar Thanks, I just pushed a new commit (slightly reworded the commit message) but the Jenkins build failed again for the same reason: https://builds.apache.org/job/PreCommit-ZOOKEEPER-github-pr-build/1990/console

lvfangmin commented:
+1 looks like the Jenkins failed due to "NoClassDefFoundError: hudson.model.Computer", is it an environment issue?

nkalmar commented:
Yes, all PR fails due to the same reason. I think there is already an INFRA ticket but I'll double check.

anmolnar commented:
@sl4mmy Would you please trigger another Jenkins build, so that we get a green one and I can commit.

sl4mmy commented:
@anmolnar Done, and it passed.  Thanks!

anmolnar commented:
Quite a simple patch to commit.
@sl4mmy Are u still around? Would you please resolve the conflict?

TWkent-spillner commented:
@anmolnar rebased & pushed.  I also added a new commit (fc9ffdb326bc6b6f4fb315da803b8eed7431b992) to handle a new error code that wasn't added to `zerror(int)` (and fixed the indentation of another case that was added to `zerror(int)`).

sl4mmy commented:
@anmolnar rebased & pushed.  I also added a new commit (fc9ffdb326bc6b6f4fb315da803b8eed7431b992) to handle a new error code that wasn't added to `zerror(int)` (and fixed the indentation of another case that was added to `zerror(int)`).

sl4mmy commented:
@anmolnar rebased & pushed.  I also added a new commit (fc9ffdb326bc6b6f4fb315da803b8eed7431b992) to handle another error code that wasn't added to `zerror(int)` (and also fixed the indentation of an error code that was added to `zerror(int)`).

However, it looks like the Travis build for s390 failed due to:

```[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for org.apache.zookeeper:parent:3.7.0-SNAPSHOT: Could not transfer artifact org.apache:apache:pom:23 from/to central (https://repo.maven.apache.org/maven2): /home/travis/.m2/repository/org/apache/apache/23/apache-23.pom.part.lock (No such file or directory) and 'parent.relativePath' points at no local POM @ line 23, column 11
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.apache.zookeeper:parent:3.7.0-SNAPSHOT (/home/travis/build/apache/zookeeper/pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for org.apache.zookeeper:parent:3.7.0-SNAPSHOT: Could not transfer artifact org.apache:apache:pom:23 from/to central (https://repo.maven.apache.org/maven2): /home/travis/.m2/repository/org/apache/apache/23/apache-23.pom.part.lock (No such file or directory) and 'parent.relativePath' points at no local POM @ line 23, column 11 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException
The command "eval mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V " failed. Retrying, 3 of 3.```

I don't think that was caused by this PR?

TWkent-spillner commented:
Ping @nkalmar @lvfangmin @breed @anmolnar - I rebased this on top of latest master earlier this morning and the build is passing.  Would be great to get this merged & closed before it's 2nd birthday (July 23), or before the 2nd anniversary of @nkalmar approving the change (July 24).  :wink:

sl4mmy commented:
Ping @nkalmar @lvfangmin @breed @anmolnar - I rebased this on top of latest master earlier this morning and the build is passing. Would be great to get this merged & closed before it's 2nd birthday (July 23), or before the 2nd anniversary of @nkalmar approving the change (July 24). :wink:

eolivelli commented:
committed to master branch ! thank you @sl4mmy
